src/librustc_ast/token.rs

```diff
@@ -740,6 +741,29 @@ impl Nonterminal {
         }
         false
     }
+
+    // See issue #74616 for details
+    pub fn ident_name_compatibility_hack(
+        &self,
+        macro_name: Symbol,
+        orig_span: Span,
```

Comment:
It's very likely that the `macro_name` can be extracted from this `orig_span` and don't need to be passed around by creating extra fields in various structures.
(And if the span is lost then the `span_to_snippet` condition won't work either.)

Reply:
I don't think it's possible to extra `wasm_bindgen` or `proc_macro_back` from `orig_span`, since it is generated before the macros are ever invoked. However, I could check for a `js-sys` or `time-macros-impl` filename, and check the name of the invoking macro (`impl_macros` or `arrays`).